monkey/infection_monkey/network/tcp_scanner.py

```diff
@@ -65,21 +63,21 @@ def _check_tcp_ports(ip: str, ports: List[int], timeout: float = DEFAULT_TIMEOUT
             logger.warning("Failed to connect to port %s, error code is %d", port, err)
 
         if len(possible_ports) != 0:
-            timeout = int(round(timeout))  # clamp to integer, to avoid checking input
             sockets_to_try = possible_ports.copy()
-            while (timeout >= 0) and sockets_to_try:
+
+            timer = Timer()
+            timer.set(timeout)
+
+            while (not timer.is_expired()) and sockets_to_try:
                 sock_objects = [s[1] for s in sockets_to_try]
 
-                _, writeable_sockets, _ = select.select([], sock_objects, [], 0)
+                _, writeable_sockets, _ = select.select([], sock_objects, [], timer.time_remaining)
```

Comment:
If this is how we're using the `timeout` argument, and the timeout for this operation (and the one on line 90) is actually `timer.time_remaining`, the docstring for the `timeout` param isn't accurate.